cppcache/src/PdxLocalWriter.hpp

```diff
@@ -110,6 +110,16 @@ class PdxLocalWriter : public PdxWriter {
     }
   }
 
+  template <typename mType>
+  void writeArrayObject(std::vector<mType> array) {
+    m_dataOutput->writeArrayLen(array.size());
+    if (array.size() > 0) {
```

Comment:
This check is redundant since the for loop will not do anything below if vector is empty.